Userland/Applications/PixelPaint/GuideTool.cpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2021, Tobias Christiansen <tobi@tobyase.de>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "GuideTool.h"
+#include "ImageEditor.h"
+#include <LibGUI/Menu.h>
+
+namespace PixelPaint {
+
+GuideTool::GuideTool()
+{
+}
+
+GuideTool::~GuideTool()
+{
+}
+
+RefPtr<Guide> GuideTool::closest_guide(const Gfx::IntPoint& point)
+{
+    auto guides = editor()->guides();
+    Guide* closest_guide = nullptr;
+    int closest_guide_distance = NumericLimits<int>::max();
+
+    for (auto& guide : guides) {
+        int relevant_position = 0;
+        if (guide.orientation() == Guide::Orientation::Horizontal)
+            relevant_position = point.y();
+        else if (guide.orientation() == Guide::Orientation::Vertical)
+            relevant_position = point.x();
+
+        auto distance = abs(relevant_position - (int)guide.offset());
+
+        if (distance < closest_guide_distance) {
+            closest_guide = &guide;
+            closest_guide_distance = distance;
+        }
+    }
```

Comment:
This makes me wish we had a `T const& max_member(Container<T> const&, TFunc&& value_mapper_fn)`...
Not a requirement for this PR or anything, just that it'd be nice to have :P